mempool/src/shared_mempool.rs

```diff
@@ -165,6 +166,22 @@ pub struct TransactionExclusion {
     pub sequence_number: u64,
 }
 
+/// Mempool's subscription to reconfiguration events
+pub struct MempoolReconfigSubscription {
+    sender: futures::channel::mpsc::Sender<usize>,
+}
+
+impl ReconfigurationEventSubscription for MempoolReconfigSubscription {
+    fn publish(&self, payload: usize) {
+        if let Err(e) = self.sender.clone().try_send(payload) {
```

Comment:
e.g. if `publish` takes `&mut self`, you can do `self.sender.try_send(payload)` w/o the clone

Reply:
what are your thoughts on dealing with backpressure? is the subscription just best-effort?

Reply:
I don't think I've seen any instance of trying to rebroadcast across the codebase - but I agree backpressure throughout the system as a whole should be eventually addressed. Maybe we can keep this as best-effort for now and iterate, since this is something we can come back to.

Reply:
in some parts of the codebase, we're using size=1 LIFO channels from e.g. network -> consensus. this is not a very general solution, but it works when you absolutely only care about the most recent message and are willing to throw away old messages if they are not handled before the next message comes in. LIFO might be relevant for some reconfig-specific subscribers who only care about the most recent reconfig notification.

Reply:
to be clear, using size=1 queue when trying to subscribe to a super common event would be a terrible idea, since there might be more than one emitted per group of transactions.

Reply:
I think at least for reconfiguration events that mempool will be interested in, they won't be so frequent - we can customize channels per subscription/use case :)